File: docs/versioned_docs/version-1.10.0/API-Reference/python-examples/api-files/download-file-v1.py
import json
import os
from pathlib import Path

import requests

base = os.environ.get("LANGFLOW_URL", "")
flow_id = os.environ.get("FLOW_ID", "")
api_key = os.environ.get("LANGFLOW_API_KEY", "")

fixtures = Path(__file__).resolve().parents[2] / "fixtures"
upload_path = Path(os.environ.get("SAMPLE_UPLOAD_FILE", str(fixtures / "sample-upload.txt")))

headers = {"accept": "application/json", "x-api-key": api_key}

upload = requests.post(
    f"{base}/api/v1/files/upload/{flow_id}",
    headers=headers,
    files={"file": (upload_path.name, upload_path.read_bytes(), "text/plain")},
    timeout=30,
)
upload.raise_for_status()
meta = upload.json()
file_name = meta["file_path"].split("/")[-1]

download = requests.get(
    f"{base}/api/v1/files/download/{flow_id}/{file_name}",
    headers=headers,
    timeout=30,
)
download.raise_for_status()

out_path = Path("downloaded_file.txt")
out_path.write_bytes(download.content)
print(json.dumps({"saved_bytes": len(download.content), "path": str(out_path), "upload": meta}))

File: docs/versioned_docs/version-1.11.0/API-Reference/javascript-examples/api-files/download-file-v1.js
const url = `${process.env.LANGFLOW_URL ?? ""}/api/v1/files/download/${process.env.FLOW_ID ?? ""}/2024-12-30_15-19-43_your_file.txt`;

const options = {
  method: 'GET',
  headers: {
    "accept": `application/json`,
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
  },
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const data = await response.arrayBuffer();
    console.log("Received binary response for downloaded_file.txt", data.byteLength);
  })
  .catch((error) => console.error(error));
